projects/components/src/table/data/table-cdk-data-source.ts

```diff
@@ -227,18 +234,19 @@ export class TableCdkDataSource implements DataSource<TableRow> {
     changedRow: StatefulTableRow | undefined
   ): Observable<StatefulTableRow[]> {
     if (changedRow !== undefined) {
-      return of(this.cachedRows).pipe(
+      return of(this.cachedData.rows).pipe(
         map(cachedRows => TableCdkRowUtil.buildRowStateChanges(cachedRows, changedRow)),
         switchMap(stateChanges => this.fetchAndAppendNewChildren(stateChanges)),
-        map(TableCdkRowUtil.removeCollapsedRows)
+        map(TableCdkRowUtil.removeCollapsedRows),
+        tap(rows => this.cacheNewData(0, rows))
```

Comment:
Line 241 to add the caching of tree rows is the only real change in the PR from what was reverted.